### PR TITLE
Add Fidelity adapter

### PR DESCRIPTION
**Pull request by onaydenov**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: 'fidelity',
  params: {
        zoneid: '27248'
  }
}
```
- contact email of the adapter’s maintainer
- [onaydenov@fidelity-media.com] official adapter submission

### CODE REVIEW

File: src/adapters/fidelity.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+var STATUS = require('../constants').STATUS;
+
+var FidelityAdapter = function FidelityAdapter() {
+  var FIDELITY_BIDDER_NAME = 'fidelity';
+  var FIDELITY_SERVER_NAME = 'x.fidelity-media.com';
+
+  function _callBids(params) {
+    var bids = params.bids || [];
+    bids.forEach(function (currentBid) {
+      if (!document.MAX_used) document.MAX_used = ',';
```
Comment (matthewlane) [inline src/adapters/fidelity.js]: I'm curious about why a variable is being attached to the `document` object like this? Could an internal function variable be used instead?

File: src/adapters/fidelity.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+      var loc = window.top !== window ? document.referrer:window.location.href;
+      loc = currentBid.params.loc || loc;
+      m3_u += '&loc=' + encodeURIComponent(loc);
+      var subid = currentBid.params.subid || 'hb';
+      m3_u += '&subid=' + subid;
+      if (document.referrer) m3_u += '&referer=' + encodeURIComponent(document.referrer);
+      if (document.context) m3_u += '&context=' + encodeURIComponent(document.context);
+      if (currentBid.params.click) {
+        document.MAX_ct0 = decodeURI(currentBid.params.click);
```
Comment (matthewlane) [inline src/adapters/fidelity.js]: Curious about attaching a variable to the `document` object here too

**Comment by onaydenov** [comment]
Hi Matt @matthewlane  ,
We'll give an update on all the points mentioned above ASAP.

File: src/adapters/fidelity.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      var m3_u = window.location.protocol + '//'+server+'/delivery/hb.php?';
+      m3_u += 'callback=window.$$PREBID_GLOBAL$$.fidelityResponse';
+      m3_u += '&requestid='+utils.getUniqueIdentifierStr();
+      m3_u += '&impid='+currentBid.bidId;
+      m3_u += '&zoneid='+currentBid.params.zoneid;
+      m3_u += '&cb='+Math.floor(Math.random()*99999999999);
+      if (document.MAX_used !== ',') m3_u += '&exclude=' + document.MAX_used;
+      m3_u += document.charset ? '&charset='+document.charset : (document.characterSet ? '&charset='+document.characterSet : '');
+
+      var loc = window.top !== window ? document.referrer:window.location.href;
```
Comment (onaydenov) [inline src/adapters/fidelity.js]: Done, wrapped window.top in a try/catch block

File: src/adapters/fidelity.js (hunk 2 above)
Comment (onaydenov) [inline src/adapters/fidelity.js]: This one has been removed also

File: src/adapters/fidelity.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        if (plugin.name.indexOf("Shockwave Flash") > -1) {
+          result = plugin.description.split("Shockwave Flash ")[1];
+        }
+      }
+    }
+    return result ? result : "";
+  }
+
+  function addBlankBidResponses(placementsWithBidsBack) {
+    var allFidelityBidRequests = $$PREBID_GLOBAL$$._bidsRequested.find(bidSet => bidSet.bidderCode === 'fidelity');
```
Comment (onaydenov) [inline src/adapters/fidelity.js]: Definition 'fidelity' replaced with variable FIDELITY_BIDDER_NAME

**Comment by onaydenov** [comment]
We'll perform additional checks to discover why we see this - "CI build failed"

**Comment by onaydenov** [comment]
Test #1276 passed successfully. It looks like everything is correct. Let me know if we should do anything else.

**Comment by onaydenov** [comment]
Hello Matt @matthewlane 
Thank you.
Please advise when prebid.js with our adapter could be available for download from the main website? Could you also let me know when adapter name may appear here http://prebid.org/download.html and at parameters description page?

**Comment by mkendall07** [comment]
@onaydenov 
Usually a few days after release. We have a release planned for Friday. Thanks

**Comment by mkendall07** [comment]
@onaydenov 
Also please add a PR here to add your bidder docs: https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders

**Comment by onaydenov** [comment]
@mkendall07 
Thank you for info. PR #145 has been submitted into proper tree.

**Pull request merged**